git_creation_and_tracking/git_creation_and_tracking.md

```diff
@@ -0,0 +1,810 @@
+
+- **Commit each chunk of changes** If you add a new section to a file, or change a certain item across multiple files, it is a good practice to commit those changes with a message describing them. If a commit message like `-m "add section on atmosphere"` or `-m "reorder paragraphs on moons from smallest to largest moon"` accurately describes your changes, make sure you commit the changes so that you can find them again later!
+
+- **Commit before making any changes that could break your code** If your code currently works are you intend to add something new that may break it, make sure you commit so that you will be able to revert to the working version if needed.
+
+- **Commit regularly** Even if you are only making very small changes, don't forget to commit at regular intervals, like before you switch tasks or end your work day.
+
+### Quiz: `add` and `commit`
+
+1. What sequence of commands would track the changes you made to `myFile.txt`?
+
+  a. `$ git add myFile.txt`
+
+  b. `$ git commit myFile.txt`
+
+  c. `$ git add -m "my short, descriptive message"`
+
+  d. `$ git commit -m "my short, descriptive message"`
+
+[( )] First run a, then b
+[(X)] First run a, then d
+[( )] First run c, then b
+[( )] First run c, then d
+***
+<div class = 'answer'>
+You must run the `git add` to tell Git to stage `myFile.txt`, then you can commit the changes and leave your short, descriptive message. We will discuss how to think of these steps in the [next section](#understanding-the-git-workflow).
+</div>
+***
+
+2. With each commit, we want to leave a short, descriptive, message describing the changes we make. Which of these would be the *best* commit message?
+
+[( )] "updates"
+[( )] "The Martian atmosphere is only 1% oxygen."
+[( )] "add a line that the Martian atmosphere is only 1% oxygen"
+[(X)] "add comments on Martian atmosphere"
+***
+<div class = "answer">
+The commit message should describe what this commit does. If you want to know the line by line changes you should use the `diff` command. The purpose of the message is so that you know what you changed without having to read all the individual changes, which is especially helpful if you edited many lines or files.
+</div>
+***
+
+## Telling Git not to track some files
+
+What if we have files that we do not want Git to track for us, like backup files created by our editor or intermediate files created during data analysis?
+
+<div class = "warning">
+It is likely that you will eventually share your repository publicly through [GitHub](github.com). When you do, everything that Git is tracking will become publicly visible. If there is an API key or anything that requires any level of secrecy, make sure you tell Git to ignore it as soon as you create the file.
```

Comment:
```suggestion
It is likely that you will eventually share your repository.  This could be sharing it privately with a team, or publicly through [GitHub](github.com). When you do, everything that Git is tracking will become visible to everyone with access to the repository.  If there is an API key, password, sensitive data, or anything that requires any level of secrecy, make sure you tell Git to ignore it as soon as you create the file.
```